[Feat] Member 엔티티 및 컨트롤러 구현

JH031 opened the pull request:
## 📣 Related Issue
<!-- 관련 이슈를 적어주세요. -->
- close #1 

## 📝 Summary
<!-- 해당 PR의 주요 작업 내용을 적어주세요 -->
- Member 엔티티 구현
- MemberController 구현
- JWT 구현
- OAuth 로그인 구현

## 🙏 Question & PR point
<!-- PR과정에서 다른 팀원이 알아야 할 사항이나 궁금증을 적어주세요 -->


## 📬 Reference
<!-- 참고한 코드의 출처를 작성해주세요 -->

JH031 committed:
feat: Member 엔티티 정의

JH031 committed:
feat: MemberController 및 DTO 구현

JH031 committed:
chore: gitignore에 환경변수 파일 추가

wien0128 commented on src/main/java/kr/co/pillguide/backend/api/member/service/MemberService.java:
```diff
@@ -0,0 +1,60 @@
+    public void updateInfo(Long memberId, MemberInfoRequestDTO request) {
+
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 회원입니다."));
```
IllegalArgumentException이 아닌 NotFoundException(404)를 던져야 추후 글로벌 예외 핸들러와 프론트에서 처리가 가능해요

JH031 committed:
chore: .env 없애고 yml로 통합

JH031 committed:
fix: Member 엔티티 수정, Gender Enum 생성, 예외처리 변경

JH031 committed:
fix: updateAdditionalInfo() 버그 수정

wien0128 committed:
feat(security, OAuth2): 스프링 시큐리티 및 OAuth2 부분 구현

- 프로젝트에 필요한 글로벌 응답 클래스 및 상태 ENUM 추가
- 스프링 시큐리티 부분 구현
- OAuth2 부분 구현

wien0128 committed:
feat(security): OAuth2 및 글로벌 예외 핸들러 등록

wien0128 committed:
feat(OAuth2): OAuth2 공급자별 사용자 정보 파싱 클래스 구현

- 각 공급자로부터 받은 사용자 정보를 형식에 맞게 파싱하여 반환
- 공급로부터 받은 사용자 정보 중 이메일이 nullable 할 수 있음

wien0128 committed:
feat(OAuth2): OAuth2 로그인 및 회원가입 로직 서비스 레이어 구현

- 중복된 이메일 혹은 이메일이 없는 경우(카카오) 예외 처리

wien0128 committed:
feat(security): 리프레쉬 토큰 분리

- 토큰 로테이트 전략을 위해 리프레쉬 토큰 별도의 엔티티로 분리
- Member와 1:N 연관관계 설정 및 편의 메서드 구현

wien0128 committed:
feat(security): UserDetails 및 OAuth2User 구현체 SecurityMember 구현

- 폼 로그인의 경우 UserDetails를 컨택스트에 저장
- 소셜 로그인의 경우 OAuth2User를 컨택스트에 저장
- security 디렉토리 위치 변경

JH031 committed:
chore: gitignore 수정

JH031 committed:
feat(security):JWT 토큰 검증 구현

JH031 committed:
feat(security): JWT 액세스/리프레시 토큰 발급 구현

JH031 committed:
feat: OAuth Google 로그인 구현

JH031 committed:
feat: 리프레쉬 토큰 재발급 로직 구현

JH031 committed:
feat: Refresh Token 저장 로직 구현, 네이버 로그인 구현

JH031 committed:
feat: 카카오 로그인 구현

JH031 committed:
feat: 로그아웃시 리프레시 토큰 삭제 구현
